src/test/clojure/clara/test_rules.clj

```diff
@@ -1183,75 +1183,85 @@
                                  [?t <- Temperature (= ?l location)]
                                  [:not [Cold (= temperature (:temperature ?t))]]]]])
 
-        s (mk-session [cold-not-match-temp] :cache false)
-        s-with-prior (mk-session [negation-with-prior-bindings] :cache false)
-        s-with-nested (mk-session [nested-negation-with-prior-bindings] :cache false)]
+        object-query (dsl/parse-query [] [[?o <- Object]])
 
-    (is (= [{}]
-           (-> s
-               (fire-rules)
-               (query cold-not-match-temp))))
+        s (mk-session [cold-not-match-temp object-query] :cache false)
+        s-with-prior (mk-session [negation-with-prior-bindings object-query] :cache false)
+        s-with-nested (mk-session [nested-negation-with-prior-bindings object-query] :cache false)
+
+        no-system-types? (fn [session]
+                           (not-any? (fn [fact] (isa? (type fact) :clara.rules.engine/system-type))
+                                     (as-> session r
```

Comment:
I was using "r" for "result".  I agree it is weird though, I just haven't thought of a better name and I like using the as-> here.  Any ideas?